packages/twenty-front/src/modules/activities/components/ActivityRichTextEditor.tsx

```diff
@@ -166,8 +198,62 @@ export const ActivityRichTextEditor = ({
         });
 
         handlePersistBody(newStringifiedBody);
+
+        // delete attachments that are no longer in the body
+        if (oldActivity.attachments.length > 0) {
```

Comment:
I feel this should be extracted to an util taking attachments + body as parameter and covered with a unit test